pages/forms/TripForm/[id].js

```diff
@@ -63,36 +62,13 @@ export default function TripDetailsForm() {
   }
 
   return (
-    <>
-      <form onSubmit={handleEditTrip}>
-        <label>
-          Country
-          <input name="country" defaultValue={country} minLength={3} required />
-        </label>
-        <label>
-          City
-          <input name="city" defaultValue={city} />
-        </label>
-        <label>
-          Title
-          <input name="title" defaultValue={title} minLength={3} required />
-        </label>
-        <label>
-          Start date
-          <input
-            name="startDate"
-            type="date"
-            defaultValue={startDate}
-            required
-          />
-        </label>
-        <label>
-          End date
-          <input name="endDate" type="date" defaultValue={endDate} />
-        </label>
-        <CreateButton>Save</CreateButton>
-      </form>
-      <CancelButton />
-    </>
+    <EditForm
+      country={country}
+      city={city}
+      title={title}
+      startDate={startDate}
+      endDate={endDate}
+      handleEditTrip={handleEditTrip}
+    />
```

Comment:
This should be `<EditForm {...trip} handleEditTrip={handleEditTrip}>`, so you pass all the info to the `EditForm` regardless of how much stuff you have inside it ^^

Inside the `EditForm` you can still do:

```js
export default function EditForm({
  title,
  location,
  ...
  handleEditTrip
}) {
  return (
    <StyledForm onSubmit={handleEditTrip}>
      <StyledLabel>
        Country
        <StyledInput
          name="country"
          defaultValue={location.country}
          minLength={3}
          required
        />
      </StyledLabel>
    </StyledForm>
  );
}
```

Reply:
Hey! We absolutely agree with you and will definitely implement these changes. We have to change the schema for that and because we have so many open PR's right now that are all dependent on the current schema it would be a huge brain mess to change it here 😅  We will refactor the schema in the next US (which will be a huge refactoring US) and afterwards adapt our code accordingly!